clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/provider/agent/KubernetesServerGroupCachingAgent.groovy

```diff
@@ -218,6 +218,42 @@ class KubernetesServerGroupCachingAgent extends KubernetesV1CachingAgent impleme
     credentials.apiAdaptor.getPods(serverGroup.namespace, serverGroup.selector)
   }
 
+  /**
+   * loaddata() need to load all pod in providers. So we load all pod in one request. That will decrease
+   * network io overhead.
+   * @param namespace
+   * @return
+   */
+  Map<String,List<Pod>> loadAllPods() {
+    def podsMap = [:]
+    namespaces.each { String namespace ->
+      def pods = credentials.apiAdaptor.getPods(namespace)
+      if (pods){
+        podsMap.put(namespace,pods)
+      }
+    }
+    return podsMap
+  }
+
+  /**
+   * If this pod is belong to specify server group return true
+   * @return
+   */
+  boolean isBelongToServerGroup(Map<String,String> podLabel,Map<String,String> serverGroupSelector){
+    def res = true
+    serverGroupSelector.each {
+      def podLabelValue = podLabel.get(it.key)
+      if (!podLabelValue) {
+        res = false
```

Comment:
Is it ok now?

Reply:
I don't think you need to set `res = false` anymore -- and at the bottom you can just `return true`, dropping `res` entirely.

Reply:
I am not very familiar with groovy. But it looks like if i use `each{}` closure ， return just take me out `each{}` iterator and this  method will always return true. So it looks like i should use `any{}` instead of `each{}` to dropping `res` . Is that ok?